tests/Unit/ControlSystem/Systems/CMakeLists.txt

```diff
@@ -0,0 +1,7 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+set(LIBRARY_SOURCES
+  ${LIBRARY_SOURCES}
+  Systems/Test_Expansion_System.cpp
```

Comment:
`Test_Expansion.cpp`

Reply:
I did this at first but got linker errors
```
CMakeFiles/Test_ControlSystem.dir/Systems/Test_Expansion.cpp.o: in function `void expand_pack<>()':
/home/knelli/Documents/research/spectre/tests/Unit/Framework/TestingFramework.hpp:35: multiple definition of `Test_ControlSystem_Test_Expansion()'; CMakeFiles/Test_ControlSystem.dir/ControlErrors/Test_Expansion.cpp.o:/home/knelli/Documents/research/spectre/src/Options/ParseOptions.hpp:676: first defined here
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[3]: *** [tests/Unit/ControlSystem/CMakeFiles/Test_ControlSystem.dir/build.make:466: lib/libTest_ControlSystem.so] Error 1
make[2]: *** [CMakeFiles/Makefile2:21710: tests/Unit/ControlSystem/CMakeFiles/Test_ControlSystem.dir/all] Error 2
```

Seems that it's mistaking the other `Test_Expansion.cpp` in the `ControlErrors` directory. Would you know how to fix this?

Reply:
Try applying this patch:
```patch
From b03b84e64ce01246bcd1c0e7a57c6dd893ee83b2 Mon Sep 17 00:00:00 2001
From: William Throwe <wtt6@cornell.edu>
Date: Wed, 23 Mar 2022 23:15:41 -0400
Subject: [PATCH] Avoid symbol collision from files with same name

---
 cmake/SpectreAddTestLibs.cmake | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/cmake/SpectreAddTestLibs.cmake b/cmake/SpectreAddTestLibs.cmake
index 6693b3daa..d7c6e875c 100644
--- a/cmake/SpectreAddTestLibs.cmake
+++ b/cmake/SpectreAddTestLibs.cmake
@@ -54,21 +54,23 @@ function(add_test_library LIBRARY FOLDER LIBRARY_SOURCES LINK_LIBS)
       )
     if(NOT ${FOUND_SPECTRE_TEST_CASE} EQUAL -1)
       # Get the name of the file without path and without extension
-      get_filename_component(
-        SOURCE_NAME
+      string(REGEX REPLACE
+        [^a-zA-Z0-9]
+        _
+        SOURCE_IDENTIFIER
         ${SOURCE_FILE}
-        NAME_WE
         )
       # We specify the macro SPECTRE_TEST_REGISTER_FUNCTION for each
       # source file in the library which is used by the TestingFramework.hpp
       # file to create a function that can be called from the main test
       # executable to get static variables to be initialized in the source
       # file.
+      set(UNIQUE_IDENTIFIER "LIB_${LIBRARY}_FILE_${SOURCE_IDENTIFIER}")
       set_source_files_properties(
         ${SOURCE_FILE}
         PROPERTIES
         COMPILE_FLAGS
-        "-D SPECTRE_TEST_REGISTER_FUNCTION=${LIBRARY}_${SOURCE_NAME}"
+        "-D SPECTRE_TEST_REGISTER_FUNCTION=${UNIQUE_IDENTIFIER}"
         )
       # We use the global "variable" (CMake doesn't really have global
       # variables but global properties are effectively the same)
@@ -83,7 +85,7 @@ function(add_test_library LIBRARY FOLDER LIBRARY_SOURCES LINK_LIBS)
         PROPERTY
         SPECTRE_TESTS_LIB_FUNCTIONS_PROPERTY)
       set(SPECTRE_TESTS_LIB_FUNCTIONS
-        "${SPECTRE_TESTS_LIB_FUNCTIONS};${LIBRARY}_${SOURCE_NAME}")
+        "${SPECTRE_TESTS_LIB_FUNCTIONS};${UNIQUE_IDENTIFIER}")
       set_property(GLOBAL
         PROPERTY
         SPECTRE_TESTS_LIB_FUNCTIONS_PROPERTY
-- 
2.34.1

```

Reply:
This works, thank you!